docs/PowerAuth-Server-1.8.0.md

```diff
@@ -31,3 +31,38 @@ values supported by the WebAuthn protocol, serialized as a JSON array.
 A new column `status_reason` has been added to the `pa_operation` table.
 It provides optional details why the status changed.
 The value should be sent in the form of a computer-readable code, not a free-form text.
+
+### Idempotency-Key of Callback URL Events
+
+Callback URL Events now include an `Idempotency-Key` in the HTTP request header. It is a unique key to recognize retries
+of the same request.
+
+### New Database Table for Callback Events Monitoring
+
+A new `pa_application_callback_event` table has been created to monitor Callback URL Events. This change introduces
+the additional benefit of setting a retry strategy for individual Callback URL Events and monitoring the state of each
+dispatched event. The table contains following columns:
+- `id` - Event identifier, used also as the `Idempotency-Key`.
+- `application_callback_id` - Reference for corresponding Callback URL record in the `pa_application_callback` table.
+- `callback_data` - Data payload of the Callback URL Event.
+- `status` - Current state of the Callback URL Event.
+- `timestamp_created` - Creation timestamp of the Callback URL Event.
+- `timestamp_last_call` - Timestamp of the last time the Callback URL Event was sent.
+- `timestamp_next_call` - Timestamp of the next scheduled time to send the Callback URL Event.
+- `timestamp_delete_after` - Timestamp after which the Callback URL Event record should be deleted from the table.
+- `attempts` - Number of dispatch attempts made for the Callback URL Event.
+
+The `pa_application_callback_event` table comes with following indices:
+- `pa_application_callback_event_status_timestamp_next_call_idx` on `(status, timestamp_next_call)`,
+- `pa_application_callback_event_timestamp_delete_after_idx` on `(timestamp_delete_after)`, and
+- `pa_application_callback_event_id_status_idx` on `(id, status)`.
```

Comment:
You are right, that in case of having an index on `(status, timestamp_next_call)` it makes no benefit to have another on `(status)` only.
The index on `(id, status)` is for the query `SELECT * FROM pa_application_callback_event WHERE id = '<id>' AND status = 'PENDING';`. Thats done by the event listener to prevent unnecessary fetching and locking a record that is not waiting to be dispatched anymore.

Reply:
id is primary key right? Then it is a unique value, so there is no need to have any additional index on PK